gen/_template/parameter_encode.tmpl

```diff
@@ -16,7 +16,7 @@ var pathParts [{{ len $.PathParts }}]string
 		if err := func() error {
     		{{- template "encode_parameter" $p -}}
 		}(); err != nil {
-			return res, errors.Wrap(err, "encode path")
+			return res, errors.Wrap(err, "encode param")
```

Comment:
This line wraps an error from **path** parameter encoder. All `errors.Wrap` calls in this template are following the same pattern: `encode ${param.in}`.

I think we can change it to something like
```
encode path parameter
```

but it should be consistent with all parameter encoding errors.